core/templates/dev/head/forms/formBuilder.js

```diff
@@ -656,6 +656,16 @@ oppia.config(['$provide', function($provide) {
     var _openCustomizationModal = function(
         customizationArgSpecs, attrsCustomizationArgsDict, onSubmitCallback,
         onDismissCallback, refocusFn) {
+      var pageContext = explorationContextService.getPageContext();
+      var componentName = customizationArgSpecs[0].name;
+      // The image component only works from the editor side.
+      // Prevent suggestors from trying to edit or add images.
+      if (componentName === 'filepath' && pageContext !== 'editor') {
```

Comment:
Also, should we hide the image in the toolbar if the page context is wrong? It seems a bit odd to show it and then immediately show an alert if the user clicks on it.